server/src/main/java/de/uftos/services/ServerService.java

```diff
@@ -77,6 +93,42 @@ public void setTimetableMetadata(TimetableMetadata timetableMetadata) {
     Server server = this.repository.findAll().getFirst();
     server.setTimetableMetadata(timetableMetadata);
 
+    this.updateTimeslots(timetableMetadata);
+
     this.repository.save(server);
   }
+
+  private void updateTimeslots(TimetableMetadata timetableMetadata) {
+    List<Timeslot> timeslots = new LinkedList<>();
+
+    for (Weekday weekday : Weekday.values()) {
+      for (int i = 0; i < timetableMetadata.getTimeslotsAmount(); i++) {
+        timeslots.add(new Timeslot(weekday, i, List.of()));
+      }
+    }
+
+    List<Timeslot> currentTimeslots = this.timeslotRepository.findAll();
+
+    if (timeslots.size() == currentTimeslots.size()) {
+      return;
+    }
+
+    if (timeslots.size() > currentTimeslots.size()) {
+      timeslots.removeIf(
+          (slot) -> slot.getSlot() < currentTimeslots.size() / Weekday.values().length);
+
+      this.timeslotRepository.saveAll(timeslots);
+      return;
+    }
+
+    List<Timetable> timetables = this.timeslotRepository.findAll().stream()
+        .flatMap(timeslot -> timeslot.getLessons().stream().map(Lesson::getTimetable)).distinct()
+        .toList();
+    this.timeslotRepository.deleteAll();
```

Comment:
Deleting all timeslots will result in all constraint instances being lost as the ids of the new timeslots are not the same as the ids of the old timeslots.
Only removed timeslots (and their instances) should be deleted.

Reply:
Hm true that sounds correct

Reply:
Should be fixed.